### DISCUSSION TITLE
Expanding Editor to multiple files

### DISCUSSION THREAD

**Discussion by TPreece101** [writer/writer-framework#134] [Ideas]
I was wondering what your thoughts are on expanding the editor to multiple files, It seems like it's currently intended for quick edits and so this might not be something you're interested in.

However, being able to do all of your required edits in the Builder would make a very complete app building experience. I've never been a fan of hopping around between windows a lot when trying to build something.

**Comment by ramedina86** [comment]
It's certainly in the roadmap to improve the editor, but no short-term plans for this. Adding the file switcher is just part of the equation; I'd feel compelled to also support autocomplete and various functionality before calling it a "real editor".

**Reply by sriver** [reply]
How about integrating something like codemirror.js (+ potential plugins) in the frontend?

**Reply by sriver** [reply]
Started walking down this path, but seems I don't fully understand the process how to incorporate external javascript funtionality into Streamsync. This would be useful both as an feature and as a tutorial, so could someone help out and write a small example / tutorial how to use codemirror (https://codemirror.net/) against a textarea field?

**Comment by TPreece101** [comment]
Yeah that makes sense, although autocomplete does sound like a much larger task 😬
